src/client/signatures/canonical_json/src/ser.rs

```diff
@@ -0,0 +1,595 @@
+                    "a": { "b": "b", "a": "a" },
+                }),
+                output: "{\"a\":{\"a\":\"a\",\"b\":\"b\"},\"b\":{\"c\":\"c\",\"d\":\"d\"}}"
+            },
+            Test {
+                test_description: "to_canonical_json_serialize_escapes_unicode_object_keys",
+                input: json!({"é": "check"}),
+                output: "{\"\\u00e9\":\"check\"}"
+            },
+            Test {
+                test_description: "to_canonical_json_serialize_multiple_nested_types",
+                input: json!({
+                    "def": "bar",
+                    "abc": json!(0.000000930258908),
+                    "ghi": json!(1000000000000000000000.0_f64),
+                    "rust": "❤",
+                    "zoo": [
+                        "zorilla",
+                        "anteater"
+                    ]
+                }),
+                output: "{\"abc\":9.30258908e-7,\"def\":\"bar\",\"ghi\":1e+21,\"rust\":\"\\u2764\",\"zoo\":[\"zorilla\",\"anteater\"]}"
+            }
+        ];
+
+        for test in &object_tests {
+            debug!("test: {}", test.test_description);
+            assert_eq!(to_canonical_json(&test.input), test.output);
+        }
+    }
+
+    #[test]
+    fn test_to_canonical_json_serialize_array_type() {
+        init();
+        let array_tests = [
+            Test {
+                test_description: "to_canonical_json_serializes_empty_array",
+                input: json!([]),
+                output: "[]"
+            },
+            Test {
+                test_description: "to_canonical_json_serializes_array_should_preserve_array_order",
+                input: json!(vec!["one", "two", "three"]),
+                output: "[\"one\",\"two\",\"three\"]"
+            },
+            Test {
+                test_description: "to_canonical_json_serializes_array_escapes_unicode_values",
+                input: json!(vec![json!({ "key": "✓" })]),
+                output: "[{\"key\":\"\\u2713\"}]"
+            },
+            Test {
+                test_description: "to_canonical_json_serializes_array_escapes_unicode_values_with_1_preceding_zeros",
+                input: json!(vec![json!({ "key": "ę" })]),
+                output: "[{\"key\":\"\\u0119\"}]"
+            },
+            Test {
+                test_description: "to_canonical_json_serializes_array_escapes_unicode_values_with_2_preceding_zeros",
+                input: json!(vec![json!({ "key": "é" })]),
+                output: "[{\"key\":\"\\u00e9\"}]"
+            },
+            Test {
+                test_description: "to_canonical_json_serialize_array_preserves_data",
+                input: json!(vec![
+                    json!({ "foo": "bar", "last_modified": "12345", "id": "1" }),
+                    json!({ "bar": "baz", "last_modified": "45678", "id": "2" }),
+                ]),
+                output: "[{\"foo\":\"bar\",\"id\":\"1\",\"last_modified\":\"12345\"},\
```

Comment:
Use [raw string literals](https://rahul-thakoor.github.io/rust-raw-string-literals/) for anything with quotes, please.

Reply:
yes